### ISSUE TITLE
PartitionKey value must be supplied for this operation.

### ISSUE THREAD

**Issue by jamsoft** [Microsoft/botbuilder-dotnet#1407]
## Version
4.2.2

## Describe the bug
Error removing state data from CosmosDB with System.InvalidOperationException - PartitionKey value must be supplied for this operation.

## To Reproduce
Any call like this could be at risk
await _accessors.ConversationState.DeleteAsync(turnContext, cancellationToken);

## Expected behavior
Should remove the state data from the database without crashing

**Comment by cleemullins** [comment]
@juanar, before we take action on this, can you investigate and make a recommendation for how to fix this?

**Comment by denscollo** [comment]
@cleemullins After doing research, we found that inside `_storage.DeleteAsync( keys, cancellationToken)` the `options` parameter is not provided to the `DeleteDocumentAsync(documentUri, options, cancellationToken)` method.  This causes the Invalid Operation Exception thrown since the `PartitionKey` is obtained from it.
This is due to the parameter `options` being marked as optional, even though according to the [documentation](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.documents.client.requestoptions.partitionkey?view=azure-dotnet#Microsoft_Azure_Documents_Client_RequestOptions_PartitionKey) it is required to execute a delete operation.

> Partition key is used to identify the target partition for this request. It **must be set** on read and delete operations for all document requests; (...)For create and update operations on documents, the partition key is optional.

In order to fix it, we changed the `DeleteAsync( keys, cancellationToken)` method to create an instance of a `RequestOptions` and pass it as a parameter to `DeleteDocumentAsync`. If it's ok we can create the PR with this fix.
